File: src/main/java/it/unibo/crabinv/Model/entity/Delta.java
package it.unibo.crabinv.Model.entity;

public enum Delta {
    DECREASE(-1),
    NO_ACTION(0),
    INCREASE(1);

    private final int value;
    Delta(int value) {
        this.value = value;
    }

    public int getValue() {
        return value;
    }
}

File: src/main/java/it/unibo/crabinv/model/entities/entity/Delta.kt
package it.unibo.crabinv.model.entities.entity

/**
 * Provides all the deltas that the entity that wishes to move should handle.
 * The selected delta affects the movement by either decreasing or increasing its coordinate in the axis.
 */
enum class Delta(
    /**
     * @return the value of delta
     */
    val value: Int,
) {
    DECREASE(-1),
    NO_ACTION(0),
    INCREASE(1),
}
